flixel/input/actions/FlxAction.hx

```diff
@@ -0,0 +1,353 @@
+	{
+		(_x != null) ? return _x : return 0;
+	}
+	
+	private function get_y():Float 
+	{
+		(_y != null) ? return _y : return 0;
+	}
+}
+
+@:allow(flixel.input.actions.FlxActionDigital, flixel.input.actions.FlxActionAnalog, flixel.input.actions.FlxActionSet)
+class FlxAction implements IFlxDestroyable
+{
+	/**
+	 * Digital or Analog
+	 */
+	public var type(default, null):FlxInputType;
+	
+	/**
+	 * The name of the action, "jump", "fire", "move", etc.
+	 */
+	public var name(default, null):String;
+	
+	/**
+	 * This action's numeric handle for the Steam API (ignored if not using Steam)
+	 */
+	private var steamHandle(default, null):Int = -1;
+	
+	/**
+	 * If true, this action has just been triggered
+	 */
+	public var fire(default, null):Bool = false;
+	
+	/**
+	 * The inputs attached to this action
+	 */
+	public var inputs:Array<FlxActionInput>;
+	
+	private var _x:Null<Float> = null;
+	private var _y:Null<Float> = null;
+	
+	private var _timestamp:Int = 0;
+	private var _check:Bool = false;
+	
+	/**
+	 * Whether the steam controller inputs for this action have changed since the last time origins were polled. Always false if steam isn't active
+	 */
+	public var steamOriginsChanged(default, null):Bool = false;
+	
+	#if steamwrap
+	private var _steamOriginsChecksum:Int = 0;
+	private var _steamOrigins:Array<EControllerActionOrigin>;
+	#end
+	
+	private function new(InputType:FlxInputType, Name:String)
+	{
+		type = InputType;
+		name = Name;
+		inputs = [];
+		#if steamwrap
+		_steamOrigins = [];
+		for (i in 0...FlxSteamController.MAX_ORIGINS)
+		{
+			_steamOrigins.push(cast 0);
+		}
+		#end
+	}
+	
+	public function getFirstSteamOrigin():Int
+	{
+		#if steamwrap
+		if (_steamOrigins == null) return 0;
+		for (i in 0..._steamOrigins.length)
+		{
+			if (_steamOrigins[i] != EControllerActionOrigin.NONE)
+			{
+				return cast _steamOrigins[i];
+			}
+		}
+		#end
+		return 0;
+	}
+	
+	public function getSteamOrigins(?origins:Array<Int>):Array<Int>
+	{
+		#if steamwrap
+		if (origins == null)
+		{
+			origins = [];
+		}
+		if (_steamOrigins != null)
+		{
+			for (i in 0..._steamOrigins.length)
+			{
+				origins[i] = cast _steamOrigins[i];
+			}
+		}
+		#end
+		return origins;
+	}
+	
+	public function removeAllInputs(Destroy:Bool = true):Void
+	{
+		var len = inputs.length;
+		for (i in 0...len)
+		{
+			var j = len - i - 1;
+			var input = inputs[j];
+			removeInput(input, Destroy);
+			inputs.splice(j, 1);
+		}
+	}
+	
+	public function removeInput(Input:FlxActionInput, Destroy:Bool = false):Void
+	{
+		if (Input == null) return;
+		inputs.remove(Input);
+		if (Destroy)
+		{
+			Input.destroy();
+		}
+	}
+	
+	public function toString():String
+	{
+		return("FlxAction(" + type + ") name:" + name);
+	}
+	
+	/**
+	 * See if this action has just been triggered
+	 * @return
+	 */
+	@:access(FlxG.game)
+	public function check():Bool
+	{
+		_x = null;
+		_y = null;
+		
+		if (_timestamp == @:privateAccess FlxG.game._total)
```

Comment:
Why is `@:privateAccess` to `FlxG.game._total` necessary here? Not a fan of that. Isn't `ticks` sufficient?